src/Microsoft.ML.Core/Data/IHostEnvironment.cs

```diff
@@ -50,6 +50,15 @@ public static IFileHandle CreateOutputFile(this IHostEnvironment env, string pat
         }
     }
 
+    [BestFriend]
+    internal interface ICancelableEnvironment
```

Comment:
Why do we have 2 interfaces? `ICancelableEnvironment` and `ICancelableHost`? Can't there just be a single interface?